ZDNA-prediction.local.2.ipynb

```diff
@@ -0,0 +1,459 @@
+    "    #    encoding='utf-8',\n",
+    "    #    format='%(asctime)s - %(levelname)s - %(name)s - %(message)s',\n",
+    "    #    level=logging.INFO\n",
+    "    #)\n",
+    "    \n",
+    "    zdnabert_model_downloader = ZdnabertModelDownloader()\n",
+    "    zdnabert_model_downloader.download_models(model_download_path)\n",
+    "    zdnabert_model_downloader.download_metas(model_download_path)\n",
+    "    \n",
+    "    zdnabert_model = ZdnabertModel(\n",
+    "        os.path.join(model_download_path, model_name),\n",
+    "        model_name=model_name,\n",
+    "        model_confidence_threshold=model_confidence_threshold,\n",
+    "        minimum_sequence_length=minimum_sequence_length,\n",
+    "        use_cuda_if_available=use_cuda_if_available,\n",
+    "    )\n",
+    "    \n",
+    "    sequence_variation_normal = SequenceVariationNormal()\n",
+    "    sequence_variation_reverse_complement = SequenceVariationReverseComplement()\n",
+    "    sequence_variations = [\n",
+    "        sequence_variation_normal,\n",
+    "        sequence_variation_reverse_complement,\n",
+    "    ]\n",
+    "    \n",
+    "    prediction_input_files = []\n",
+    "    for uploaded_item_key in tqdm(uploaded_items.keys(), 'preparing'):\n",
+    "        uploaded_item_path = uploaded_items[uploaded_item_key]\n",
+    "        \n",
+    "        prediction_input_file = PredictionInputFileFromFilesystem(\n",
+    "            uploaded_item_key,\n",
+    "            uploaded_item_path,\n",
+    "        )\n",
+    "        prediction_input_files.append(prediction_input_file)\n",
+    "        \n",
+    "    prediction_input = PredictionInput(\n",
+    "        zdnabert_model,\n",
+    "        prediction_input_files,\n",
+    "        sequence_variations,\n",
+    "    )\n",
+    "    prediction_inputs = [\n",
+    "        prediction_input,\n",
+    "    ]\n",
+    "\n",
+    "    prediction_result_formatter_bed_file = PredictionResultFormatterBedFile()\n",
+    "    \n",
+    "    prediction_runner = PredictionRunner()\n",
+    "\n",
+    "    for prediction_result in prediction_runner.run(prediction_inputs, progress_bar=tqdm):\n",
+    "        bed_file_name_seq = prediction_result_formatter_bed_file.file_name(prediction_result)\n",
+    "        \n",
+    "        print(bed_file_name_seq)\n",
+    "        #print(bed_file_name)\n",
+    "\n",
+    "        bed_file_seq_handler = open(os.path.join(output_path, bed_file_name_seq), 'w')\n",
+    "        \n",
+    "        #print('track name=\"{name}\" priority=1'.format(name=model_params_as_string))\n",
+    "        #print('track name=\"{name}\" priority=2'.format(name=seq_name))\n",
+    "        for line in prediction_result_formatter_bed_file.format(prediction_result):\n",
+    "            #print(line)\n",
+    "            bed_file_seq_handler.write(\"%s\\n\" % line)\n",
+    "\n",
+    "        bed_file_seq_handler.close()\n",
+    "        #print()"
```

Comment:
combined bed file implementation is currently missing when compared to the original notebook

@AlistairNGarratt, was the combined bed file actually useful or can we omit it for the re-implementation?

Reply:
I think the combined .bed file is useful as it reduces the work needed later in spreadsheets to combine the normal and rev-comp files manually. I think it would be good to have a one line header at the top of the combined .bed file, includeing the important information, i.e. search parameters, sequence name (I think you have this included already). The row that lies between the ‘normal’ and ‘rev-comp’ entries in the older code should be removed in the combined .bed file.

Reply:
done in 93f3786bd44f5356d8b259d13075691dba32ff69